### PR TITLE
2024 07 09 ma uniswap v3

### PR DESCRIPTION
**Pull request by TobyMeller**
## Checks

By submitting this for review, I'm confirming I've done the following:
- [x] made this PR as small as possible
- [x] unit-tested any new functionality
- [x] linked any relevant issues or PRs

### CODE REVIEW

**Comment by TobyMeller** [comment]
Need to consolidate down to one strat

File: strategies/BidAskSpreadMA/consolidating-MA-buyside-wmaticuusdt.rain [hunk 1]
```diff
@@ -47,8 +47,12 @@ scenarios:
         orderbook: polygon-wmatic
         bindings:
           # Ask for now, registry in future.
-          uniswap-words: 0x2382e861cF4F47578aC29B50944b3b445577aF74
+          uniswap-words: 0xbc609623F5020f6Fc7481024862cD5EE3FFf52D7
```
Comment (Siddharth2207) [inline strategies/BidAskSpreadMA/consolidating-MA-buyside-wmaticuusdt.rain]: This is the updated uniswap parser.

File: strategies/BidAskSpreadMA/consolidating-MA-buyside-wmaticuusdt.rain [hunk 2]
```diff
@@ -159,16 +165,16 @@ moving-average-length: price-average-time,
 price-average: uniswap-v3-twap-output-ratio(
     buy-token sellback-token 
     price-average-time 0
-    poolfee
+    uniswap-v3-factory uniswap-v3-init-code poolfee
```
Comment (Siddharth2207) [inline strategies/BidAskSpreadMA/consolidating-MA-buyside-wmaticuusdt.rain]: updated uniswap parser needs factory and init-code literal can we have the same for rest of them?

File: strategies/BidAskSpreadMA/consolidating-MA-buyside-wmaticuusdt.rain [hunk 3]
```diff
@@ -178,12 +184,11 @@ current-time: block-timestamp(),
 
 /*set conditions*/
 buycondition: every(
-  is-zero(get("trimeme-26-has-buy-binary")) 
+  is-zero(get("trimeme-24-has-buy-binary"))
```
Comment (Siddharth2207) [inline strategies/BidAskSpreadMA/consolidating-MA-buyside-wmaticuusdt.rain]: can we have pinned keys at the top of the .rain file?

File: strategies/BidAskSpreadMA/consolidating-MA-buyside-wmaticuusdt.rain [hunk 4]
```diff
@@ -178,12 +184,11 @@ current-time: block-timestamp(),
 
 /*set conditions*/
 buycondition: every(
-  is-zero(get("trimeme-26-has-buy-binary")) 
+  is-zero(get("trimeme-24-has-buy-binary")) 
   greater-than(current-price range-floor) 
   less-than(current-price range-ceiling)),
 
 /*price params*/
```
Comment (Siddharth2207) [inline strategies/BidAskSpreadMA/consolidating-MA-buyside-wmaticuusdt.rain]: expect `price-needed`, the rest of the elemets of the price params are irrelevant to the execution of the order, so they should be on the plottables stack within the plot binding and not in prod.

**Review by Siddharth2207 — APPROVED** [review]

**Pull request merged**